### PR TITLE
Ivan/emscripten fetch

### PR DESCRIPTION
**Pull request by ivan-cukic**

### CODE REVIEW

**Review by RalphSteinhagen — CHANGES_REQUESTED** [review]
@ivan-cukic first of all nice work! I know, the devil is in the details/build system.


I had a first read-through and the following things should IMO be addressed:
 * [ ] there should be a build script and instructions for the combined native/WASM build. This doesn't have to be a single CMakeList.txt but a bash or similar script would work equally
 * [x] the WASM REST client must be unit-testable not locally as well as the CI, similar to the the native one using a stripped down HTTP client, notably testing the (via the `node`[.js] interpreter that comes with the emsdk):
   * [x] GET/SET API + failure handling
   * [ ] subscription handling, for this PR at least one of the following options (the other later):
     * [x] using long-polling -> long-term default
     * [ ] using server-sent events -> OK for local use but does not scale w.r.t. crossing network domain boundaries
     * [ ] perhaps: WebSockets -- only if this is easily done since WS does not particularly scale well for many clients
 * [ ] the WASM REST client should support at least the JSON and YAS-based serialiser protocols and automatically (de-)serialise from/to aggregate types
   * how to handle transmission/decoding/other errors ...?
 * [ ] since the transmission performance is very critical, we would need a ping-pong throughput test to ensure that we can achieve bandwidth > 250 MB/s
 * [ ] there should be a user-level example (in ../concepts/.. ?) that illustrate the WASM/native REST client API. E.h. using cmd line arguments/`node` to query an opencmw or any other arbitrary web service. N.B. Subscription probably needs a taylored service.
 
Notably, the complexity should be kept as low/small as possible so that novice users can easily read/adopt/use the client API... naming and interfaces are hard, but I think we need to spend a bit more design on this.

File: scripts/native_run_test.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+parser.add_argument('--server-build-dir')
+parser.add_argument('--test-name')
+args = parser.parse_args()
+
+os.environ["DISABLE_REST_HTTPS"] = "1"
+
+rest_command = "exec " + args.server_build_dir + "/concepts/majordomo/MajordomoRest_example " + os.getcwd() + "/"
+print("Starting server " + rest_command)
+rest_server = subprocess.Popen(rest_command, stdout=subprocess.PIPE, shell=True)
```
[the remark this answers is not in the document]
Comment (ivan-cukic) [reply scripts/native_run_test.py:22]: Not a public API

File: scripts/native_run_test.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+time.sleep(2)
+
+error_present = False
+
+try:
+    test_command = "exec ./build_native/" + args.test_name
+    print("Running test command " + test_command)
+    test_process = subprocess.Popen(test_command)
```
[the remark this answers is not in the document]
Comment (ivan-cukic) [reply scripts/native_run_test.py:31]: Not a public API

**Review by RalphSteinhagen — APPROVED** [review]
Large and extensive commit! Thanks @ivan-cukic for the changes and for integrating the unit test for the WebAPI/WASM related things. :+1: 

As discussed, the integration of the WASM tests would be done via a separate PR by @wirew0rm. Also, as discussed this would be the low-level MdpMessage based API. For the user-facing API we should also integrate the domain-object-based serialisers... (another PR).

We may iterate once this is used/integrated into Giulios UI...

**Pull request merged**